pkg/client/src/app/pages/applications/analysis-wizard/custom-rules.tsx

```diff
@@ -32,15 +32,71 @@ import {
 import { useFilterState } from "@app/shared/hooks/useFilterState";
 import { Rule } from "@app/api/models";
 import { NoDataEmptyState } from "@app/shared/components";
+import { IReadFile } from "./components/add-custom-rules";
 
 import "./custom-rules.css";
+import { UseFormGetValues } from "react-hook-form";
+import { IFormValues } from "./analysis-wizard";
 
-export const CustomRules: React.FunctionComponent = () => {
+interface ICustomRules {
+  customRulesFiles: IReadFile[];
+  setValue: (files: IReadFile[]) => void;
+}
+
+export const CustomRules: React.FunctionComponent<ICustomRules> = ({
+  customRulesFiles,
+  setValue,
+}) => {
   const [isAddCustomRulesModalOpen, setCustomRulesModalOpen] =
     React.useState(false);
-  const [files, setFiles] = React.useState<File[]>([]);
 
-  const rules: Rule[] = [];
+  const rules = React.useMemo(() => {
+    const getRules = (file: IReadFile) => {
+      if (!file.data) return [];
+
+      let source = "";
+      let target = "";
+      let rulesCount = 0;
+
+      const payload = atob(file.data.substring(21));
+      const parser = new DOMParser();
+      const xml = parser.parseFromString(payload, "text/xml");
+
+      const ruleSets = xml.getElementsByTagName("ruleset");
+
+      if (ruleSets && ruleSets.length > 0) {
+        const metadata = ruleSets[0].getElementsByTagName("metadata");
+
+        if (metadata && metadata.length > 0) {
+          const sources = metadata[0].getElementsByTagName("sourceTechnology");
+          if (sources && sources.length > 0) source = sources[0].id;
+
+          const targets = metadata[0].getElementsByTagName("targetTechnology");
+          if (targets && targets.length > 0) target = targets[0].id;
+        }
+
+        const rulesGroup = ruleSets[0].getElementsByTagName("rules");
+        if (rulesGroup && rulesGroup.length > 0)
+          rulesCount = rulesGroup[0].getElementsByTagName("rule").length;
+      }
+
+      return [
+        {
+          name: file.fileName,
+          source: source,
+          target: target,
+          total: rulesCount,
+        },
+      ] as Rule[];
+    };
+
+    let rules: Rule[] = [];
+    customRulesFiles.forEach((file) => {
+      if (file.data) rules = [...rules, ...getRules(file)];
```

Comment:
Can we use curly brackets for if statements just for consistency's sake.

Reply:
This is proper js/ts syntax and don't think there is an eslint rule to enforce that anyway.

Reply:
Ok. I understand that it is valid - I just believe that we should be consistent with either using them or not. I would argue that it makes code more readable to have curly brackets. Semicolons are not syntactically required, but we add them for legibility.

Reply:
I agree we could argue about it for a long time as this is quite opinionated. 
We could be consistent by agreeing to have no brackets when is only one statement.
It effectively makes the code more readable which is paramount along with consistency.
Please let me know if you being consistent that way. In which case we can use brackets for conditional statements that are constituted of single statement.